File: LeetCode/0012. Integer to Roman/solution.cpp
/**
 * 3999 / 3999 test cases passed.
 * Runtime: 20 ms
 * Memory Usage: 9.2 MB 
 */
class Solution {
public:
    const vector<pair<int, string>> romans = {
        {1000, "M"},
        {900, "CM"},
        {500, "D"},
        {400, "CD"},
        {100, "C"},
        {90, "XC"},
        {50, "L"},
        {40, "XL"},
        {10, "X"},
        {9, "IX"},
        {5, "V"},
        {4, "IV"},
        {1, "I"},
    };
    string intToRoman(int num) {
        string ans = "";
        for (auto& [value, symbol]: romans) {
            while (num - value >= 0) {
                ans += symbol;
                num -= value;
            }
        }
        return ans;
    }
};

File: LeetCode/0012. Integer to Roman/solution.py
"""
3999 / 3999 test cases passed.
Runtime: 52 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def intToRoman(self, num: int) -> str:
        values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
        symbols = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I']
        ans = ''
        for value, symbol in zip(values, symbols):
            while num >= value:
                num -= value
                ans += symbol
        return ans
